### ISSUE TITLE
ASP.NET Core SetCompatibilityVersion(CompatibilityVersion.Version_2_2) Breaks LightInject

### ISSUE THREAD

**Issue by nystrup** [seesharper/LightInject#482]
A completely boilerplate ASP.NET Core 2.2 API project doesn't work with LightInject if you run under IIS Express (i.e., the default F5 behavior in Visual Studio) or IIS.

```
System.InvalidOperationException: Attempt to create a disposable instance without a current scope.
   at LightInject.PerRequestLifeTime.TrackInstance(Scope scope, IDisposable disposable) in C:\projects\lightinject\src\LightInject\LightInject.cs:line 6229
   at LightInject.PerRequestLifeTime.GetInstance(Func`1 createInstance, Scope scope) in C:\projects\lightinject\src\LightInject\LightInject.cs:line 6213
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.AspNetCore.Routing.Matching.DfaMatcherFactory.CreateMatcher(EndpointDataSource dataSource)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.InitializeCoreAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

Running the application as a console app does, however, work.

Another workaround is to entirely remove the SetCompatibilityVersion() call in the default ConfigureServices method or explicitly pass a Version_2_0 argument.  This does, however, prevent you from taking advantage of all features of more resent versions of ASP.NET Core.

**Comment by seesharper** [comment]
@nystrup We are aware of this issue and we are working with a fix for this

**Comment by seesharper** [comment]
@nystrup Fixed by #486 For AspNet.Core application we recommend to bump `LightInject.Microsoft.AspNetCore.Hosting` to version 1.2.0
